frontend/src/components/StockData/StockData.jsx

```diff
@@ -13,14 +15,14 @@ export default function StockData({ currentStock }) {
             try {
                 setIsLoading(true);
 
-                const databaseResponse = await axios.get(`http://localhost:3000/stocks/${currentStock}`);
+                const databaseResponse = await axios.get(`http://localhost:3000/stocks/${ticker}`);
                 const stockExistsInDatabase = databaseResponse.data ? true : false;
 
                 if (stockExistsInDatabase) {
                     setStockData(databaseResponse.data);
                 } else {
-                    const stockOverviewUrl = `https://www.alphavantage.co/query?function=OVERVIEW&symbol=${currentStock}&apikey=${ALPHA_API_KEY}`;
-                    const stockPriceUrl = `https://finnhub.io/api/v1/quote?symbol=${currentStock}&token=${FINNHUB_API_KEY}`;
+                    const stockOverviewUrl = `https://www.alphavantage.co/query?function=OVERVIEW&symbol=${ticker}&apikey=${ALPHA_API_KEY}`;
+                    const stockPriceUrl = `https://finnhub.io/api/v1/quote?symbol=${ticker}&token=${FINNHUB_API_KEY}`;
```

Comment:
You can look into `URL` API, which provides builders, making it cleaner and simpler to add on various search parameters to a base URL